frappe/utils/doctor.py

```diff
@@ -33,118 +13,47 @@ def purge_pending_tasks(event=None, site=None):
 	events but of the all event type, ie. the ones that are enqueued every five
 	mintues and would any leave daily, hourly and weekly tasks
 	"""
-	r = get_redis_conn()
-
-	event_tasks = []
-	for app in frappe.get_all_apps(True):
-		all_events = frappe.get_hooks(app_name=app).get('scheduler_events', {})
-		if event:
-			event_tasks += all_events.get(event) or []
-		else:
-			# tasks of all events
-			for tasks in all_events.values():
-				event_tasks += tasks or []
-
-	count = 0
+	purged_task_count = 0
+	for queue in get_queue_list():
+		q = get_queue(queue)
+		purged_task_count += q.count
```

Comment:
This should also be `per site` if site is passed or `per event` if event is passed